pkgdb/src/pkgdb/write.cc

```diff
@@ -213,28 +216,8 @@ PkgDb::PkgDb( const nix::flake::LockedFlake & flake, std::string_view dbPath )
 void
 PkgDb::connect()
 {
-  /* The `locking_mode` pragma acquires an exclusive write lock the first time
-   * that the database is written to and only releases the lock once the
-   * database connection is closed. We make a write as soon as possible after
-   * opening the connection to establish the write lock. After the
-   * `RETRY_WHILE_BUSY` call returns we should be the only process able to write
-   * to the database, though other processes mays still read from the database
-   * (this is why we must use `EXCLUSIVE` transactions, which prevent reads as
-   * well).
-   *
-   * It could be the case that this database hasn't been initialized yet, so we
-   * can't write to an existing table. Instead we just write to a dummy table.
-   * It's unclear whether setting a pragma value like `appliation_id` counts as
-   * a write, so we create a table instead.*/
-  static const char * acquire_lock = R"SQL(
-  BEGIN EXCLUSIVE TRANSACTION;
-  CREATE TABLE IF NOT EXISTS _lock (foo int);
-  COMMIT TRANSACTION
-  )SQL";
   this->db.connect( this->dbPath.string().c_str(),
                     SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE );
-  sqlite3pp::command cmd( this->db, acquire_lock );
```

Comment:
I didn't know about `sql_busy_timeout` until @tomberek brought it up.  But that looks like what we're reproducing here, but within SQL.  Seems like a good use of a builtin feature that should just catch all our SQL calls universally and without us having to maintain that code.  It's possible there's hidden optimizations within their built in implementations.  Is there a specific reason that wasn't used ( aside from not knowing about it like I would have done :)  )?

Reply:
Yeah I hadn't heard of it either.
I can try using that here which should help us avoid the explicit retry blocks.